### ISSUE TITLE
SA1629 should ignore <c> & <code>

### ISSUE THREAD

**Issue by af4jm** [DotNetAnalyzers/StyleCopAnalyzers#2784]
`/// <param name="message">Format string of the log message in message template format. Example: <c>"User {User} logged in from {Address}."</c></param>`
results in a warning for SA1629, not ending documentation with a period.  Using `<code>` in place of `<c>` has the same result
removing the <c> tag fixes the warning

**Comment by sharwell** [comment]
📝 `<code>` is a block level element (#2761). Only `<c>` would be part of the discussion here.

**Comment by af4jm** [comment]
thinking about it a little more, I'd say that any subtag (c, code, para, etc.) that has an open & a close with something in between, should be ignored by all rules about the content of a doc comment.
I phrased that the way I did because self-closing tags (see, paramref, & typeparamref) should NOT be ignored because they render text rather than formatting containing text

**Comment by sharwell** [comment]
> ```csharp
> /// <param name="message">Format string of the log message in message template format. Example: <c>"User {User} logged in from {Address}."</c></param>
> ```

I'm not sure I agree with the example here. The `.` in this example is part of a string literal (an opaque character as opposed to the end of a sentence).

**Comment by vweijsters** [comment]
I would say **wontfix** as well.

**Comment by martincostello** [comment]
I have a similar example here after updating to 1.1.118:

```xml
<param name="language">The optional language to use, in the format <code>languagecode2-country/regioncode2</code>, such as <code>en-GB</code>.</param>
```

SA1629 is firing that the comment does not end in a `.` where `<code></code>` is used mid-sentence.

**Comment by sharwell** [comment]
@martincostello that is covered by #2761
